runtime/doc/lua.txt

```diff
@@ -2353,4 +2353,20 @@ parents({start})                                            *vim.fs.parents()*
     Return: ~
         (function) Iterator
 
+
+==============================================================================
+Lua module: secure                                                *lua-secure*
```

Comment:
I see that `vim.readsecure()` https://github.com/neovim/neovim/pull/20925 was closed but don't see a rationale. Why introduce a new module for this? Do we plan to have many more functions in `vim.secure`? 

The current functionality at least seems to be covered by a parameter on `vim.read({secure=true, path=...})`. Otherwise `vim.readsecure()` works, if the interface is expected to be different than `vim.read()`.

Reply:
See @ii14's [comment](https://github.com/neovim/neovim/pull/20956#discussion_r1014756332). The thinking was there may be other useful APIs that could fall under the "secure" umbrella (e.g. an interface to add or remove items from the trust database).

Reply:
"vim.secure" is quite a lofty name for this :) Maybe `vim.localrc` or `vim.exrc`. 

Not trying to bikeshed, but thinking of `trust()` and looking at https://github.com/ii14/exrc.vim/blob/master/doc/exrc.txt and related plugins, clearly the direction is to support the features of these "local vimrc". `vim.secure` isn't the right name for this concept.

Reply:
vimrcs are the most obvious use case for this, but it could be used for anything really, not necessarily even vim script or lua.

Reply:
>The current functionality at least seems to be covered by a parameter on vim.read({secure=true, path=...}). Otherwise vim.readsecure() works, if the interface is expected to be different than vim.read().

I do think that the "readsecure" interface might be different enough to warrant being separate from a more generic `read` function. For example, we might want to expose the ability to specify the prompt text, or modify some parameters of how the file is displayed when the user wants to view it (e.g. setting the buffer filetype). None of these would make sense for a generic `read` function.

Re: `vim.secure.read()` vs `vim.readsecure()`: I don't have a dog in that fight, but I'd prefer we just pick something and not bikeshed too long. It sounds like @ii14 already has some ideas on other functions that would fit in a "secure" module. We also have plenty of time until the next release cycle if we want to just throw it under `vim.readsecure()` for now and refactor into a separate module later on when we have more use cases.